scripts/partition_manager/partition_manager.rst

```diff
@@ -96,34 +99,83 @@ placement: dict
+
+   .. code-block:: yaml
+
+      # This partition spans, or contains, partition_1 through partition_n,
+      # in any order:
+      container_partition_name:
+        span: [partition_1, partition_2, ..., partition_n]
+
+   The list elements are interpreted as the set of potential partitions in the container, which Partition Manager may place on flash in any order. For example, ``partition_2`` could be placed before ``partition_1``.
+
+   Non-existent partitions are removed from the ``span`` list before processing, and partitions with empty ``span`` lists are removed altogether (unless filled via the ``inside`` property described :ref:`below <partition_manager_inside>`).
+
+   If Partition Manager is forced to place a partition that is not declared in the ``span`` list between two partitions that are in the list, the configuration is unsatisfiable and therefore invalid.
+   See :ref:`Span property example 1 <partition_manager_span_ex1>` for an example invalid configuration.
+
+   If the value is a dict with key ``one_of``, the semantics are equivalent to a list which contains the first existing partition in the ``one_of``-list:
+
+   .. code-block:: yaml
+
+      # If partition_1 is defined, this container spans just that partition.
+      # Otherwise, if partition_2 is defined, it spans just that partition,
+      # and so on:
+      container_partition_name:
+        span: {one_of: [partition_1, partition_2, ..., partition_n]}
+
+   If the value is a string, it is interpreted as a list with one item:
 
+   .. code-block:: yaml
+
+      # The following examples are equivalent:
 
-span: list OR dict: list
-   If the value type is a list, this property lists which partitions this partition should span across.
-   If the value type is a dict with key ``one_of``, the semantics are equivalent
-   to having a list with only the first existing partition in the ``one_of``-list.
-   A string formatted value is interpreted as a single item list.
-   Partitions with this property are container partitions.
-   Therefore, this property cannot be used together with the ``placement`` property.
+      container_partition_name:
+        span: foo
 
-   Non-existing partitions are removed from the ``span`` list before processing, and partitions with empty ``span`` lists are removed altogether (unless filled via ``inside``).
+      container_partition_name:
+        span: [foo]
 
    .. note::
       You can specify configurations with an ambiguous ordering (see the following examples).
-      However, different versions of the script might produce a different ordering for such configurations, and the Partition Manager might fail to find a solution even if one is theoretically possible.
+      Different versions of the Partition Manager script may either produce different partition orders for such configurations, or fail to find a solution even if one is possible.
```

Comment:
```suggestion
      Different versions of the Partition Manager script might produce different partition orders for such configurations, or fail to find a solution even if one is possible.
```

Reply:
I'm using "may" to indicate a requirement level in the sense of [RFC2119](https://tools.ietf.org/html/rfc2119), since that's a familiar engineering usage.

I'm not aware of standards documents which use "might" in the same way, so when I read this document, I wondered if this note was saying that there were problems with Partition Manager being unpredictable in an undesirable way. Changing to "may" makes it clearer to me personally that this ambiguity is by design, which I believe is the intent.

Should I really change "may" back to "might"?

I'm also not sure why "either" should be removed. Seems like a correct usage of "either / or": "either produce [...], or fail to find a solution". Is this a house style?

Reply:
"May" is permission, "might" is possibility. So if you want to point out that it is allowed that different versions produce different results, "may" is correct. But I think you want to point out that it could happen that there are different results, and there it must be "might".

"Either" should be removed because it introduces two options - either A or B. But I'm pretty sure there's an option C here, namely that the different versions actually give the same result.

Reply:
I believe PM "may" (in the sense of permission produce different orderings or  fail to find a valid solution even if one exists as described here -- @hakonfam, can you confirm?

This is a point I find confusing, which I'm trying to make clearer with this patch. By switching to "may," I am explicitly trying to make the documentation say say that if that happens, it's not a bug.

I will drop "either" -- I see your point that it's usually (though I think [not always](https://en.wikipedia.org/wiki/Exclusive_or#Exclusive_%22or%22_in_English)) used as an exclusive-or.